packages/next/src/build/index.ts

```diff
@@ -3705,6 +3710,14 @@ export default async function build(
 
       await shutdownPromise
     })
+  } catch (e) {
+    const telemetry: Telemetry | undefined = traceGlobals.get('telemetry')
+    if (telemetry) {
+      telemetry.record(
+        eventBuildFailed({ bundler: getBundlerForTelemetry(isTurbopack) })
```

Comment:
Do we want to capture class of failure e.g. syntax error or others also maybe duration here as well so we know how fast it failed?